src/netbase.cpp

```diff
@@ -246,7 +246,7 @@ struct timeval MillisToTimeval(int64_t nTimeout)
  *
  * @note This function requires that hSocket is in non-blocking mode.
  */
-bool static InterruptibleRecv(char* data, size_t len, int timeout, SOCKET& hSocket)
+bool static InterruptibleRecv(uint8_t* data, size_t len, int timeout, SOCKET& hSocket)
```

Comment:
In the upstream PR `data` is cast back to a `char*` in the body of this function:
```ssize_t ret = recv(hSocket, (char*)data, len, 0); // Optimistically try the recv first```
Why is that not done here?

Reply:
It's unnecessary, as @daira could explain better. I think it was done in the upstream PR to reduce the worry about the PR introducing a bug and to perform the fix more covertly, or maybe due to a misunderstanding.